Transports/com.mlapi.contrib.transport.facepunch/Runtime/FacepunchTransport.cs

```diff
@@ -0,0 +1,354 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using MLAPI.Logging;
+using MLAPI.Transports.Tasks;
+using Steamworks;
+using Steamworks.Data;
+using UnityEngine;
+
+namespace MLAPI.Transports.Facepunch
+{
+    using SocketConnection = Steamworks.Data.Connection;
+
+    public class FacepunchTransport : NetworkTransport, IConnectionManager, ISocketManager
+    {
+        private ConnectionManager connectionManager;
+        private SocketManager socketManager;
+        private Dictionary<ulong, Client> connectedClients;
+        private Dictionary<NetworkChannel, SendType> channelSendTypes;
+        private Queue<ConnectionEvent> eventQueue;
+
+        [SerializeField] private List<TransportChannel> channels = new List<TransportChannel>();
+
+        [Space]
+        [Tooltip("The Steam App ID of your game. Technically you're not allowed to use 480, but Valve doesn't do anything about it so it's fine for testing purposes.")]
+        [SerializeField] private uint steamAppId = 480;
+
+        [Tooltip("The Steam ID of the user targeted when joining as a client.")]
+        [SerializeField] public ulong targetSteamId;
+
+        [Header("Info")]
+        [ReadOnly]
+        [Tooltip("When in play mode, this will display your Steam ID.")]
+        [SerializeField] private ulong userSteamId;
+
+        private LogLevel LogLevel => NetworkManager.Singleton.LogLevel;
+
+        private class Client
+        {
+            public SteamId steamId;
+            public SocketConnection connection;
+        }
+
+        private struct ConnectionEvent
+        {
+            public NetworkEvent type;
+            public ulong clientId;
+            public NetworkChannel channel;
+            public ArraySegment<byte> payload;
+            public float receiveTime;
+        }
+
+        #region MonoBehaviour Messages
+
+        private void Awake()
+        {
+            try
+            {
+                SteamClient.Init(steamAppId, false);
+            }
+            catch (Exception e)
+            {
+                if (LogLevel <= LogLevel.Error)
+                    Debug.LogError($"[{nameof(FacepunchTransport)}] - Caught an exeption during initialization of Steam client: {e}");
+            }
+            finally
+            {
+                StartCoroutine(InitSteamworks());
+            }
+        }
+
+        private void FixedUpdate()
+        {
+            SteamClient.RunCallbacks();
```

Comment:
https://wiki.facepunch.com/steamworks/Installing_For_Unity#running

Garry Newman actually recommends doing RunCallbacks() in the Update() Event, not sure if this is relevant though

Reply:
I'm actually not a fan of doing any Steamworks related stuff other than networking inside the transport. In our game for example we have a dedicated SteamManager that takes care of these things.

Reply:
I prefer it if the transport is more plug-and-play, so it just immediatly works and doesn't need any other script to function. Also, he says "every frame or so", and I thought it would be a good idea to call it a constant amount of times per second.